benchmarking/commons/hpo_main_common.py

```diff
@@ -36,6 +37,177 @@
 PostProcessingType = Callable[[Tuner], Any]
 
 
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ("yes", "true", "t", "y", "1"):
+        return True
+    elif v.lower() in ("no", "false", "f", "n", "0"):
+        return False
+    else:
+        raise ArgumentTypeError("Boolean value expected.")
+
+
+@dataclass
+class Parameter:
+    name: str
+    type: Any
+    help: str
+    default: Any
+    required: bool = False
+
+
+class ConfigDict(dict):
```

Comment:
sure, we dont need the inheritance. I thought it would be more understandable but maybe you are right that there might be some uncesessary edge cases.